### PR TITLE
GAWB-2334: one-click sharing when importing config

### PR DESCRIPTION
**Pull request by MattPutnam**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/endpoints.cljs [hunk 1]
```diff
@@ -582,32 +586,26 @@
...
+(defn post-agora-namespace-acl [namespace config?]
+  {:path (str "/" (if config? "configurations" "methods") "/" namespace "/permissions")
    :method :post})
 
 
-(defn delete-agora-entity [config? ns n sid]
+(defn delete-agora-entity [config? {:keys [namespace name snapshotId]}]
   {:path (let [base (if config? "configurations" "methods")]
-           (str "/" base "/" ns "/" n "/" sid))
+           (str "/" base "/" namespace "/" name "/" snapshotId))
```
Comment (ahaessly) [inline src/cljs/main/broadfcui/endpoints.cljs]: in the other agora endpoints, we use a let to set base, we just make the comparison in-line. maybe change fort consistency

File: src/cljs/main/broadfcui/page/workspace/method_configs/synchronize.cljs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+     (swap! state dissoc :grant-error)
+     (endpoints/call-ajax-orch
+      {:endpoint (endpoints/persist-agora-entity-acl false (:method props))
+       :payload (if (:customize? @state)
+                  (mapv (fn [[user level]] {:user user :role (string/upper-case level)}) @locals)
+                  (mapv (fn [user] {:user user :role "READER"}) (:users props)))
+       :headers utils/content-type=json
+       :on-done (fn [{:keys [success? get-parsed-response]}]
+                  (if success?
+                    ((:dismiss props))
```
Comment (ahaessly) [inline src/cljs/main/broadfcui/page/workspace/method_configs/synchronize.cljs]: why is this in double parens?

**Review by ahaessly — APPROVED** [review]
👍

**Review by dmohs — APPROVED** [review]
I can't be thorough since I'm in meetings all day but the code looks great. Thanks for using the new modals :)

File: src/cljs/main/broadfcui/page/workspace/method_configs/synchronize.cljs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                    :onChange #(swap! locals assoc user (.. % -target -value))}
+                   ["Reader" "Owner" "No access"])]]])
+             (:users props))]]])
+   :-grant-permission
+   (fn [{:keys [props state locals]}]
+     (swap! state dissoc :grant-error)
+     (endpoints/call-ajax-orch
+      {:endpoint (endpoints/persist-agora-entity-acl false (:method props))
+       :payload (if (:customize? @state)
+                  (mapv (fn [[user level]] {:user user :role (string/upper-case level)}) @locals)
```
Comment (davidangb) [inline src/cljs/main/broadfcui/page/workspace/method_configs/synchronize.cljs]: Any users who were specified to have "no access" should just be removed from the payload altogether. In other words, send:
```
[
  { "user" : "alice@example.com", "role" : "READER"},
  { "user" : "carol@example.com", "role" : "OWNER"}
]
```
instead of:
```
[
  { "user" : "alice@example.com", "role" : "READER"},
  { "user" : "bob@example.com", "role" : "NO ACCESS"},
  { "user" : "carol@example.com", "role" : "OWNER"}
]
```
first, it's a barely more efficient payload. But more importantly, sending an explicit "no access" ACL for a user triggers a weirdness in agora, where agora actually inserts the no-access ACL and that user appears in the permissions list (as "no access").

File: src/cljs/main/broadfcui/page/workspace/method_configs/synchronize.cljs [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                  (mapv (fn [user] {:user user :role "READER"}) (:users props)))
+       :headers utils/content-type=json
+       :on-done (fn [{:keys [success? get-parsed-response]}]
+                  (if success?
+                    ((:dismiss props))
```
Comment (davidangb) [inline src/cljs/main/broadfcui/page/workspace/method_configs/synchronize.cljs:86]: when I actually run this, my brain wanted to see an "ok, I set permissions as you asked" acknowledgment instead of the modal just disappearing. I'm fine handling that as a follow-on.

**Review by davidangb — COMMENTED** [review]
Two comments inline. Additionally:
* does this affect/break any automation?
* could you work with QA to write up a (quick) test plan / set of conditions to test, outlining some of the corner cases you found?

File: src/cljs/main/broadfcui/page/workspace/method_configs/synchronize.cljs (hunk 4 above)
Comment (MattPutnam) [reply src/cljs/main/broadfcui/page/workspace/method_configs/synchronize.cljs:86]: Yeah, I think we need some sort of notification widget for this, and for several other places where a modal simply disappears on success.

**Review by davidangb — APPROVED** [review]
👍 on the code
